src/overlays/actors/ovl_En_Hy/z_en_hy.c

```diff
@@ -29,71 +44,1023 @@ const ActorInit En_Hy_InitVars = {
+
+static EnHyUnknownStruct6 D_80A725A4[] = {
+    { 36.0f, { 0.0f, 0.0f, 600.0f }, 0.01f, 0x06, 30.0f },     { 40.0f, { -100.0f, 0.0f, 400.0f }, 0.01f, 0x06, 30.0f },
+    { 22.0f, { 0.0f, 0.0f, -200.0f }, 0.01f, 0x06, 30.0f },    { 20.0f, { -100.0f, 0.0f, 0.0f }, 0.01f, 0x06, 30.0f },
+    { 22.0f, { 0.0f, 0.0f, 0.0f }, 0.01f, 0x06, 30.0f },       { 21.0f, { 0.0f, 0.0f, 0.0f }, 0.01f, 0x06, 30.0f },
+    { 25.0f, { -100.0f, 0.0f, 600.0f }, 0.01f, 0x06, 30.0f },  { 28.0f, { -100.0f, 0.0f, 800.0f }, 0.01f, 0x06, 30.0f },
+    { 17.0f, { 0.0f, 0.0f, 700.0f }, 0.01f, 0x06, 30.0f },     { 18.0f, { 0.0f, 0.0f, 100.0f }, 0.01f, 0x06, 30.0f },
+    { 18.0f, { 0.0f, 0.0f, -200.0f }, 0.01f, 0x06, 30.0f },    { 17.0f, { 0.0f, 0.0f, 700.0f }, 0.01f, 0x06, 30.0f },
+    { 21.0f, { 0.0f, 0.0f, -300.0f }, 0.01f, 0x06, 30.0f },    { 20.0f, { 0.0f, 0.0f, -200.0f }, 0.01f, 0x06, 30.0f },
+    { 18.0f, { -200.0f, 0.0f, -200.0f }, 0.01f, 0x06, 30.0f }, { 27.0f, { -100.0f, 0.0f, 800.0f }, 0.01f, 0x06, 30.0f },
+    { 19.0f, { 400.0f, 0.0f, 0.0f }, 0.01f, 0x04, 30.0f },     { 19.0f, { 400.0f, 0.0f, 0.0f }, 0.01f, 0x06, 30.0f },
+    { 40.0f, { -100.0f, 0.0f, 400.0f }, 0.01f, 0x06, 30.0f },  { 17.0f, { 0.0f, 0.0f, 700.0f }, 0.01f, 0x06, 30.0f },
+    { 20.0f, { 0.0f, 0.0f, -200.0f }, 0.01f, 0x06, 30.0f },
+};
+
+s32 func_80A6F5B0(EnHy* this, GlobalContext* globalCtx) {
+    u8 i2 = D_80A722D8[this->actor.params & 0x7F].unk_0;
+    u8 i1 = D_80A722D8[this->actor.params & 0x7F].unk_1;
+    u8 i0 = D_80A722D8[this->actor.params & 0x7F].unk_6;
+
+    this->unk_198 = Object_GetIndex(&globalCtx->objectCtx, D_80A72010[i0].objectId);
+    if (this->unk_198 < 0) {
+        return false;
+    }
+
+    this->unk_197 = Object_GetIndex(&globalCtx->objectCtx, D_80A72010[i1].objectId);
+    if (this->unk_197 < 0) {
+        return false;
+    }
+
+    this->unk_196 = Object_GetIndex(&globalCtx->objectCtx, D_80A71F50[i2].objectId);
+    if (this->unk_196 < 0) {
+        return false;
+    }
+
+    return true;
+}
+
+s32 func_80A6F6C0(EnHy* this, GlobalContext* globalCtx) {
+    if (!Object_IsLoaded(&globalCtx->objectCtx, this->unk_198)) {
+        return false;
+    }
+
+    if (!Object_IsLoaded(&globalCtx->objectCtx, this->unk_197)) {
+        return false;
+    }
+
+    if (!Object_IsLoaded(&globalCtx->objectCtx, this->unk_196)) {
+        return false;
+    }
+
+    return true;
+}
+
+s32 func_80A6F744(EnHy* this, GlobalContext* globalCtx) {
+    this->unk_199 = Object_GetIndex(&globalCtx->objectCtx, OBJECT_OS_ANIME);
+    if (this->unk_199 < 0) {
+        return false;
+    }
+
+    return true;
+}
+
+s32 func_80A6F790(EnHy* this, GlobalContext* globalCtx) {
+    if (!Object_IsLoaded(&globalCtx->objectCtx, this->unk_199)) {
+        return false;
+    }
+
+    return true;
+}
+
+void func_80A6F7CC(EnHy* this, GlobalContext* globalCtx, s32 itemId) {
+    this->unk_260 = itemId;
+    func_8002F434(&this->actor, globalCtx, itemId, this->actor.xzDistToLink + 1.0f,
```

Comment:
To avoid any confusion with actual "items" (`ITEM_` values), I suggest naming this argument `getItemId`.
```suggestion
void func_80A6F7CC(EnHy* this, GlobalContext* globalCtx, s32 getItemId) {
    this->unk_260 = getItemId;
    func_8002F434(&this->actor, globalCtx, getItemId, this->actor.xzDistToLink + 1.0f,
```